packages/backend/src/open_payments/payment/outgoing/service.test.ts

```diff
@@ -428,6 +431,566 @@ describe('OutgoingPaymentService', (): void => {
+            assetCode: 'USD',
+            assetScale: 9
+          }
+          const grant = new Grant({
+            active: true,
+            grant: uuid(),
+            access: [
+              {
+                type: AccessType.OutgoingPayment,
+                actions: [AccessAction.Create, AccessAction.Read],
+                identifier: `${Config.publicHost}/${accountId}`,
+                interval: `R0/${start.toISOString()}/P1M`,
+                limits: sendAmount
+                  ? { sendAmount: amount }
+                  : { receiveAmount: amount }
+              }
+            ]
+          })
+          await expect(
+            outgoingPaymentService.create({ ...options, grant })
+          ).resolves.toEqual(OutgoingPaymentError.InsufficientGrant)
+        }
+      )
+      test.each`
+        sendAmount | failed   | description
+        ${true}    | ${false} | ${'sendAmount'}
+        ${false}   | ${false} | ${'receiveAmount'}
+        ${true}    | ${true}  | ${'sendAmount'}
+        ${false}   | ${true}  | ${'receiveAmount'}
+      `(
+        'fails if limit was already used up - $description',
+        async ({ sendAmount, failed }): Promise<void> => {
+          const grantAmount = {
+            value: BigInt(200),
+            assetCode: sendAmount
+              ? quote.asset.code
+              : quote.receiveAmount.assetCode,
+            assetScale: sendAmount
+              ? quote.asset.scale
+              : quote.receiveAmount.assetScale
+          }
+          const grant = new Grant({
+            active: true,
+            grant: uuid(),
+            access: [
+              {
+                type: AccessType.OutgoingPayment,
+                actions: [AccessAction.Create, AccessAction.Read],
+                identifier: `${Config.publicHost}/${accountId}`,
+                interval: `R0/${start.toISOString()}/P1M`,
+                limits: {
+                  sendAmount: sendAmount ? grantAmount : undefined,
+                  receiveAmount: sendAmount ? undefined : grantAmount
+                }
+              }
+            ]
+          })
+          const paymentAmount = {
+            ...grantAmount,
+            value: BigInt(190)
+          }
+          const firstPayment = await createOutgoingPayment(deps, {
+            accountId,
+            receiver: `${
+              Config.publicHost
+            }/${uuid()}/incoming-payments/${uuid()}`,
+            sendAmount: sendAmount ? paymentAmount : undefined,
+            receiveAmount: sendAmount ? undefined : paymentAmount,
+            grant,
+            validDestination: false
+          })
+          assert.ok(firstPayment)
+          if (failed) {
+            await firstPayment
+              .$query(trx)
+              .patch({ state: OutgoingPaymentState.Failed })
+
+            jest
+              .spyOn(accountingService, 'getTotalSent')
+              .mockResolvedValueOnce(sendAmount ? BigInt(188) : BigInt(188 * 2))
```

Comment:
I tried changing this to:
```suggestion
              .mockResolvedValueOnce(sendAmount ? BigInt(11) : BigInt(188 * 2))
```
but the payment is created successfully and the test fails.

I think this is due to `estTotalAvailableSendAmount` not being updated:
https://github.com/interledger/rafiki/pull/366/files#diff-c7f2b9ad5c5975c96f5355c09c8f95fcf5da3c117b1f073ff2c2ec2ee9d0b338R465
https://github.com/interledger/rafiki/pull/366/files#diff-c7f2b9ad5c5975c96f5355c09c8f95fcf5da3c117b1f073ff2c2ec2ee9d0b338R465

Reply:
I think that is because 11 doesn't use up the grant and a second payment can still be successfully created.

Reply:
You're right. I couldn't keep my amounts straight.
The test passes for `78` but fails for `77`, which is correct with the grant send amount limit of `200` and the payment send amount of `123`. :heavy_check_mark: